Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -635,10 +635,29 @@ public final class ManifestLoader: ManifestLoaderProtocol {
                         // This provides some safety against arbitrary code execution when parsing manifest files.
                         // We only allow the permissions which are absolutely necessary.
                         if self.isManifestSandboxEnabled {
-                            let cacheDirectories = [self.databaseCacheDir, moduleCachePath].compactMap{ $0 }
-                            let strictness: Sandbox.Strictness = toolsVersion < .v5_3 ? .manifest_pre_53 : .default
                             do {
-                                cmd = try Sandbox.apply(command: cmd, strictness: strictness, writableDirectories: cacheDirectories)
+                                var sandbox = SandboxProfile()
+
+                                // Allow writing inside the temporary directories.
+                                sandbox.pathAccessRules.append(.writable(try AbsolutePath(validating: "/tmp")))
+                                sandbox.pathAccessRules.append(.writable(try localFileSystem.tempDirectory))
```

Comment:
should we consider changing `FileSystem::tempDirectory` to return an array that includes `/tmp` instead of adding it manually everywhere? alternatively, maybe a function in this area of the code?

Reply:
I'm not sure I'd want to change the `tempDirectory` function itself, since `/tmp` is discouraged (in favour of the per-user one that `NSTemporaryDirectory()` returns), and so making it an array adds complexity without any gain (IMO).

I agree that we'd want to avoid the duplication.  In this case, I think the hope is that we can remove the `/tmp` hardcoding when we set `TMPDIR` in a follow-on PR, which I would actually want to do as an option on `Process` (along with adding the sandbox as an option).  So the duplication here is fairly temporary, I hope.

Reply:
how about something like 

```
// deprecate - for compatibility reason 12/2022
FileSystem._legacyTempDirecotry
```

Reply:
Having discussed this further offline, I think a improvement for this PR would be to follow Tom's suggestion of encapsulating the `/tmp` part into a `FileSystem.legacyTemporaryDirectory` or something similar.  That way it's clear that it's disfavored compared with the regular temporary directory, but at least the hardcoding is mitigated.  Then I also want to open that `TSC.Process` PR to add the support for the sandbox and for passing through access to the temporary directory there.

Reply:
Having thought some more about this, maybe the right thing to do here is to reintroduce the option for whether or not "temporary directories are writable" as a `SandboxProfile` option in its own right (it was removed in a previous change in order to simplify the sandbox profile).  I'm not all that keen on making `FileSystem` have two separate notions of a temporary directory, which seems conceptually confusing to most clients.  The inclusion of `/tmp` here is really just for backward compatibility and so it seems unfortunate to add semi-deprecated API to `FileSystem` for it.

The construction of these paths could then be pulled into a single place in the sandbox profile, and clients just get to choose whether or not to enable writing to temporary directories.

Reply:
ok